build/mac/BUILD.gn

```diff
@@ -242,16 +254,111 @@ if (skip_signing) {
       shell_script,
       unsigned_dmg,
     ]
-    outputs = [ "${root_out_dir}/$mac_signing_output_prefix/$brave_dmg" ]
+    outputs = [ "${root_out_dir}/$mac_signed_output_prefix/$brave_dmg" ]
     args = [
       rebase_path(shell_script, root_build_dir),
       rebase_path(unsigned_dmg),
-      rebase_path("$root_out_dir/$brave_dmg"),
+      rebase_path("$root_out_dir/$mac_signed_output_prefix/$brave_dmg"),
       keychain_db,
       mac_signing_identifier,
       "-r=designated => identifier \"$chrome_product_full_name\" and certificate leaf = H\"$mac_signing_identifier\"",
     ]
 
     deps = [":create_dmg"]
   }
+
+  group("finalize_dmg_pkg") {
+    deps = [
+      ":copy_dmg_pkg"
+    ]
+  }
+
+  copy("copy_notarize_script") {
+
+    sources = [ "notarize_dmg_pkg.py" ]
+
+    outputs = [
+      "$packaging_dir/{{source_file_part}}"
+    ]
+
+  }
+
+  action("notarize_dmg_pkg") {
+    if (is_official_build) {
+      development_arg = ""
+      if (brave_channel == "") {
+        provisioning_profile = "//brave/build/mac/release.provisionprofile"
+      } else {
+        provisioning_profile = "//brave/build/mac/${brave_channel}.provisionprofile"
+      }
+    } else {
+      development_arg = "--development"
+      provisioning_profile = "//brave/build/mac/dummy.provisionprofile"
+    }
+
+    script = "$packaging_dir/notarize_dmg_pkg.py"
```

Comment:
(per Zoom review) Can we do the `rebase_path` on this too?

Reply:
I tried a few invocations of this, `rebase_path("$packaging_dir/notarize_dmg_pkg.py")` a well as `rebase_path("notarize_dmg_pkg.py", "$packaging_dir")` but in both cases I get the error:

```
python ../../brave/build/mac/notarize_dmg_pkg.py --keychain /Users/jenkins/Library/Keychains/signing.keychain-db --identity 392F0BA65B348B1FA2FA028AA11752A01B8481A5 --notary-user dev@brave.com --notary-password ejzp-hczf-ddbk-xrwp  --dmg /Users/jenkins/temp/mbacchi-notarize-test-build/brave-browser-8dv56k/src/out/Release/signed/Brave\ Browser\ Nightly.dmg --pkg /Users/jenkins/temp/mbacchi-notarize-test-build/brave-browser-8dv56k/src/out/Release/signed/Brave\ Browser\ Nightly.pkg --signed /Users/jenkins/temp/mbacchi-notarize-test-build/brave-browser-8dv56k/src/out/Release/signed --outdir /Users/jenkins/temp/mbacchi-notarize-test-build/brave-browser-8dv56k/src/out/Release/notarized --pkgdir /Users/jenkins/temp/mbacchi-notarize-test-build/brave-browser-8dv56k/src/out/Release/Brave\ Browser\ Nightly\ Packaging --mac_provisioning_profile /Users/jenkins/temp/mbacchi-notarize-test-build/brave-browser-8dv56k/src/brave/build/mac/nightly.provisionprofile
Traceback (most recent call last):
  File "../../brave/build/mac/notarize_dmg_pkg.py", line 27, in <module>
    from signing import config, commands, model, notarize, pipeline, signing  # noqa: E402
ImportError: No module named signing
```

This shows that its running the python script from the `root_build_dir` rather than the `$packaging_dir`. I think the first argument to the python interpreter in an `action()` target is the `script` but having spaces in the python script path may not be supported. I haven't seen any examples of this in the Brave or Chromium codebase after a little searching.

Reply:
well crap. @bridiver given this, I'd like to propose we keep the underscores in right now- but create a follow up issue and prioritize it on the Devops board

Reply:
I tried a few invocations of this:
`rebase_path("$packaging_dir/notarize_dmg_pkg.py")` a well as `rebase_path("notarize_dmg_pkg.py", "$packaging_dir")` but in both cases I get the error:

```
python ../../brave/build/mac/notarize_dmg_pkg.py ...
Traceback (most recent call last):
  File "../../brave/build/mac/notarize_dmg_pkg.py", line 27, in <module>
    from signing import config, commands, model, notarize, pipeline, signing  # noqa: E402
ImportError: No module named signing
```

This shows that its running the python script from the `root_build_dir` rather than the `$packaging_dir`. I think the first argument to the python interpreter in an `action()` target is the script but having spaces in the python script path may not be supported. I haven't seen any examples of this in the Brave or Chromium codebase after a little searching.